pkg/client/unversioned/deployment.go

```diff
@@ -94,12 +94,13 @@ func (c *deployments) Update(deployment *extensions.Deployment) (result *extensi
 }
 
 // Watch returns a watch.Interface that watches the requested deployments.
-func (c *deployments) Watch(label labels.Selector, field fields.Selector, resourceVersion string) (watch.Interface, error) {
+func (c *deployments) Watch(label labels.Selector, field fields.Selector, opts api.ListOptions) (watch.Interface, error) {
 	return c.client.Get().
 		Prefix("watch").
 		Namespace(c.ns).
 		Resource("deployments").
-		Param("resourceVersion", resourceVersion).
+		Param("resourceVersion", opts.ResourceVersion).
```

Comment:
I'm not sure I understand - you still want to pass api.ListOptions to Watch method, and convert it into VersionedParams here? What is the goal of it?
I though that client here is by design unversioned and we use by design only unversioned objects here.

[No there wasn't any discussion about it].